Content/Items/Misc/Accessories.RhythmicResonator.cs

```diff
@@ -0,0 +1,199 @@
+
+namespace StarlightRiver.Content.Items.Misc
+{
+    public class RhythmicResonator : SmartAccessory
+    {
+        public override string Texture => AssetDirectory.MiscItem + Name;
+
+        public RhythmicResonator() : base("Rhythmic Resonator", "Attack in rhythm with your weapon to gradually increase damage and knockback\nDisables autoswing") { }
+
+        public override void Load()
+        {
+            StarlightItem.CanAutoReuseItemEvent += PreventAutoReuse;
+            StarlightItem.UseItemEvent += UseItemEffects;
+        }
+
+        public override void Unload()
+        {
+            StarlightItem.CanAutoReuseItemEvent -= PreventAutoReuse;
+            StarlightItem.UseItemEvent -= UseItemEffects;
+        }
+
+        public override void SafeSetDefaults()
+        {
+            Item.value = Item.sellPrice(gold: 1);
+            Item.rare = ItemRarityID.Green;
+        }
+
+        public override void SafeUpdateEquip(Player Player)
+        {
+            Player.GetModPlayer<RhythmicResonatorModPlayer>().equipped = true;
+        }
+
+        private bool? PreventAutoReuse(Item Item, Player Player)
+        {
+            if (Equipped(Player))
+                return false;
+
+            return null;
+        }
+
+        private bool? UseItemEffects(Item Item, Player Player)
+        {
+            var modPlayer = Player.GetModPlayer<RhythmicResonatorModPlayer>();
+            if (Equipped(Player))
+            {
+                if ((modPlayer.RhythmTimer > -2 && modPlayer.RhythmTimer < 3) || modPlayer.bufferedInput) // five frame window plus buffer
+                {
+                    modPlayer.bufferedInput = false;
+                    modPlayer.flashTimer = 20;
+                    modPlayer.RhythmStacks++;
+                    modPlayer.ResetTimer = Utils.Clamp((int)((Item.useTime * (1f - (Player.GetTotalAttackSpeed(Item.DamageType) - 1f))) * 2f), 30, 180); //set the reset timer for the stacks to triple the items use time, clamped at 3 seconds
+                    SoundEngine.PlaySound(SoundID.MenuTick with { Volume = 1.15f, Pitch = -0.2f, PitchVariance = 0.15f }, Player.Center);
+
+                    if (Main.myPlayer == Player.whoAmI) //only spawn dust on client
+                        for (int i = 0; i < 15; i++)
+                        {
+                            Dust.NewDustPerfect(Main.MouseWorld + (Vector2.One * 10) + Main.rand.NextVector2Circular(2.5f, 2.5f), ModContent.DustType<Dusts.GlowFastDecelerate>(), Main.rand.NextVector2Circular(1.75f, 1.75f), 0, Color.White, 0.7f);
+                        }
+                }
+
+                float speed = CombinedHooks.TotalUseTime(Item.useTime, Player, Item) + 1; //one tick of leeway
+                modPlayer.RhythmTimer = (int)speed;
+                modPlayer.MaxRhythmTimer = (int)speed;
+            }
+            return null;
+        }
+    }
+
+    public class RhythmicResonatorModPlayer : ModPlayer //needs a modplayer for the UI
```

Comment:
I question this reasoning -- as UI is a client-side only phenomenon there should be no issue tracking timers such as these from the UIState class. Additionally the damage modifying functions here could follow the delegation pattern of other SmartAccessories with the 'combo' counter being tracked in the item itself.

Reply:
I may need to add a static version of SmartAccessory.GetEquippedInstance to ease this transition